correios/models/posting.py

```diff
@@ -272,34 +271,86 @@ def __init__(self,
         Package.validate(package_type, width, height, length, diameter, service, weight)
 
         if len(sequence) != 2 or sequence[0] > sequence[1]:
-            raise InvalidPackageSequenceError("Package must be a tuple with 2 elements: (number, total)")
+            raise exceptions.InvalidPackageSequenceError("Package must be a tuple with 2 elements: (number, total)")
 
         self.package_type = package_type
-        self.width = math.ceil(width)  # cm
-        self.height = math.ceil(height)  # cm
-        self.length = math.ceil(length)  # cm
-        self.diameter = math.ceil(diameter)  # in cm
-        self.weight = math.ceil(weight)  # in grams
+
+        self.real_width = width  # cm
+        self.real_height = height  # cm
+        self.real_length = length  # cm
+        self.real_diameter = diameter  # in cm
+        self.real_weight = diameter  # in grams
         self.sequence = sequence
         self.service = service
 
     @property
-    def volumetric_weight(self):
+    def width(self):
+        return max(MIN_WIDTH, int(math.ceil(self.real_width)))
+
+    @width.setter
+    def width(self, width):
+        Package._validate_dimension("width", width, MAX_WIDTH)
+        self.real_width = width
+
+    @property
+    def height(self):
+        return max(MIN_HEIGHT, int(math.ceil(self.real_height)))
+
+    @height.setter
+    def height(self, height):
+        Package._validate_dimension("height", height, MAX_HEIGHT)
+        self.real_height = height
+
+    @property
+    def length(self):
+        return max(MIN_LENGTH, int(math.ceil(self.real_length)))
+
+    @length.setter
+    def length(self, length):
+        Package._validate_dimension("length", length, MAX_LENGTH)
+        self.real_length = length
+
+    @property
+    def diameter(self):
+        return max(MIN_DIAMETER, int(math.ceil(self.real_diameter)))
+
+    @diameter.setter
+    def diameter(self, diameter):
+        Package._validate_dimension("diameter", diameter, MAX_DIAMETER)
+        self.real_diameter = diameter
+
+    @property
+    def weight(self):
+        return int(math.ceil(self.real_weight))
+
+    @weight.setter
+    def weight(self, weight):
+        if weight <= 0:
+            raise exceptions.InvalidMinPackageWeightError("Invalid weight {!r}g".format(weight))
+
+        if self.service and weight > self.service.max_weight:
```

Comment:
Two things here:
* As I could see, it's possible to exist a Package without a service. The business rules allow a Package without service? Seems weird.
* I think the exception message could be improved putting the service name on it. `Max weight exceeded [...] for service xpto`. If you accept it, there's another equal message in `validate()`